packages/react-day-picker/src/contexts/SelectMultiple/SelectMultipleProvider.tsx

```diff
@@ -22,14 +21,23 @@ export function SelectMultipleProvider({
   initialProps: DayPickerProps;
   children: React.ReactNode;
 }): JSX.Element {
-  const [selectedDays, setSelectedDays] = useControllablePropState(
-    isDayPickerMultiple(initialProps)
-      ? {
-          value: initialProps.selected,
-          defaultValue: initialProps.defaultSelected
-        }
-      : { defaultValue: undefined }
-  );
+  const [selectedDays, setSelectedDays] = useControlledValue<
+    Date[] | undefined
+  >(initialProps.defaultSelected, initialProps.selected);
+
+  const modifiers: SelectMultipleModifiers = {
+    selected: [],
+    disabled: []
+  };
+
+  if (!isDayPickerMultiple(initialProps)) {
+    const contextValue = { selected: undefined, modifiers };
+    return (
+      <SelectMultipleContext.Provider value={contextValue}>
+        {children}
+      </SelectMultipleContext.Provider>
+    );
+  }
```

Comment:
This is what I was talking about in my earlier comment:
```
export function SelectMultipleProvider({
  initialProps,
  children
}: {
  initialProps: DayPickerProps;
  children: React.ReactNode;
}): JSX.Element {
  if (!isDayPickerMultiple(initialProps)) {
    return (
      <SelectMultipleContext.Provider value={EMPTY_SELECT_MULTIPLE_CONTEXT}>
        {children}
      </SelectMultipleContext.Provider>
    );
  }
  return <SelectMultipleProviderInternal
    initialProps={initialProps}
    children={children}
  />
}

const EMPTY_SELECT_MULTIPLE_CONTEXT = {
  selected: undefined,
  modifiers: {
    selected: [],
    disabled: []
  }
}

function SelectMultipleProviderInternal({
  initialProps,
  children
}: {
  // vvv This is a safe cast because of type predicates in isDayPickerMultiple
  initialProps: DayPickerBase & DayPickerMultiple;
  children: React.ReactNode;
}): JSX.Element {
  // vvv It means we don't need the unsafe cast here:
  const [selectedDays, setSelectedDays] = useControlledValue(
    initialProps.defaultSelected, initialProps.selected);

  const handleDayClick: DayClickEventHandler = (day, modifiers, e) => {
    initialProps.onDayClick?.(day, modifiers, e);

```

Reply:
I'll put up a PR

Reply:
Here's a complete view compared to master: https://github.com/gpbl/react-day-picker/pull/1285

Reply:
Here's a view of just my refactor: https://github.com/gpbl/react-day-picker/pull/1286

Reply:
Thanks will follow up there!